pages/docs/user-docs/build-docker-module.md

```diff
@@ -0,0 +1,43 @@
+---
+title: docker bootstrap module
+sidebar: user_docs
+permalink: build-docker-module
+folder: docs
+toc: false
+---
+
+This module allows you to build a container from an existing container hosted on Docker Hub or another Docker registry. 
+
+{% include toc.html %}
+
+## Overview
+You can use an existing container image as your "base," and then add customization. For example, instead of using the `debootstrap` module to start an Ubuntu container, you might use the official Ubuntu container from Docker Hub as a base.  You could do the same with CentOS, Debian, Arch, Suse, Alpine, BusyBox, etc.  Or maybe you want to build a container that uses CUDA and cuDNN to leverage the GPU, but ou don't want to install from scratch.  You can start with one of the `nvidia/cuda` containers and install your software on top of that.
+
+## Keywords
+```
+Bootstrap: docker
+```
+The **Bootstrap** keyword is always mandatory. It describes the bootstrap module to use.
+```
+From: /username/container:tag
+```
+The **From** keyword is mandatory.  It specifies the container to use as a base. In the case of some official registries (ubuntu for example) the `username` is optional.  The `tag` is also optional and will default to `latest`
+```
+Registry: http://custom_registry
+```
+The **Registry** keyword is optional.  It will default to `docker://index.docker.io`.
+```
+Token: yes
+```
+The **Token** keyword is optional.  It defaults to `no`.  In some cases the Docker API will request an authentication token which can be generated on the fly by using this keyword.  
```

Comment:
I didn't implement this - does this work?

Reply:
I have no idea.  I just copied it from [here](https://github.com/singularityware/singularityware.github.io/commit/c98435be707df76ec494a3c3ee614ce80368be62#diff-0ba67df7f15ebecf97344aea586a7165R84) (which I assumed you wrote).  It looks like maybe you removed this section though?  I'm confused.

Reply:
I don't really remember - let's remove it to be safe and worst case scenario, we investigate and re-add if needed. Most of the time I wouldn't want to encourage a user to put any kind of credential in a file, and it doesn't make sense given that they expire pretty quickly (and we get them automatically).

Reply:
I'm just going to remove the bit about the token keyword and refer users to the Singularity and Docker page instead.